Core/GDCore/IDE/ProjectExposerHelper.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+#ifndef GDCORE_PROJECTEXPOSERHELPER_H
+#define GDCORE_PROJECTEXPOSERHELPER_H
+namespace gd {
+class Project;
+class String;
+class EventsFunctionsExtension;
+class EventsFunction;
+class EventsBasedBehavior;
+class EventsBasedObject;
+class ArbitraryEventsWorker;
+class ArbitraryEventsWorkerWithContext;
+class ArbitraryFunctionsWorker;
+class ArbitraryObjectsWorker;
+class ArbitraryEventBasedBehaviorsWorker;
+class ArbitrarySharedDataWorker;
+} // namespace gd
+
+namespace gd {
+
+/**
+ * \brief Expose a subset of the project to workers.
+ */
+class GD_CORE_API ProjectExposerHelper {
+public:
+  /**
+   * \brief Call the specified worker on all events of the project (layout,
+   * external events, events functions...)
+   *
+   * This should be the preferred way to traverse all the events of a project.
+   */
+  static void ExposeProjectEvents(gd::Project &project,
+                                  gd::ArbitraryEventsWorker &worker);
+
+  /**
+   * \brief Call the specified worker on all events of the project (layout,
+   * external events, events functions...)
+   *
+   * This should be the preferred way to traverse all the events of a project.
+   */
+  static void ExposeProjectEvents(gd::Project &project,
+                                  gd::ArbitraryEventsWorkerWithContext &worker);
+
+  /**
+   * \brief Call the specified worker on all events of the events based
+   * behavior
+   *
+   * This should be the preferred way to traverse all the events of an events
```

Comment:
`event-based behavior` if I follow the convention in the next methods documentation